src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -22,7 +22,7 @@ public abstract class CollectionTableValue<T> : TableValue
         private readonly IReadOnlyList<T> _sourceIndex; // maybe null. supports index. 
         private readonly IReadOnlyCollection<T> _sourceCount; // maybe null. supports count;
 
-        public CollectionTableValue(RecordType recordType, IEnumerable<T> source)
+        internal CollectionTableValue(RecordType recordType, IEnumerable<T> source)
```

Comment:
>internal [](http://example.com/codeflow?start=8&length=8)

why are we demoting the visibility?

Reply:
[https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1012](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1012)